.github/workflows/dagster.yml

```
@@ -59,7 +59,7 @@ on:
 jobs:
   run-docker-workflow:
     # only run on synchronize PR event, meaning something has been pushed to the PR branch
-    if: ${{ github.event.action != 'closed' || github.event.action == 'push' }}
+    if: (github.event.action != 'closed' || github.event.action == 'push') && github.event.action != 'unlabeled'
```

If we want to be really explicit then this would be:
```suggestion
    if: contains(fromJSON('["opened", "synchronize", "reopened", "push"]'), github.event.action) || (github.event_name == 'pull_request' && github.event.action == 'labeled' && github.event.label.name == 'dev-env')
```

Run docker build if:
* PR is opened, sync'd or reopened or if there is a push to the main branch, or
* `dev-env` label is added to the pull request

Ref: [pull_request labeled event payload](https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads?actionType=labeled#pull_request)